code/datums/weather/weather_types/floor_is_lava.dm

```diff
@@ -30,4 +30,6 @@
 		return
 	if(!L.client) //Only sentient people are going along with it!
 		return
+	if(issilicon(L))
+		return
```

Comment:
Can you put this above the first check? It shouldn't even bother looking for objects in `loc` if the mob is a silicon.

Reply:
oh yeah definitely put this first